src/org/jetbrains/plugins/scala/caches/HydraArtifactsCache.scala

```diff
@@ -0,0 +1,41 @@
+package org.jetbrains.plugins.scala.caches
+
+import java.util.concurrent.ConcurrentHashMap
+
+import org.jetbrains.plugins.scala.project.Platform
+import org.jetbrains.plugins.scala.project.template.Downloader
+
+/**
+  * @author Maris Alexandru
+  */
+object HydraArtifactsCache {
+  private val cache = new ConcurrentHashMap[(String, String), Seq[String]]()
+  private val SPLIT_REGEX = "\\[info\\] \\* Attributed\\(|\\)"
+  private val ARTIFACTS_REGEX = "\\[info\\] \\* Attributed\\(.*\\)"
+  private val HYDRA_REGEX = ".*hydra_\\d+\\.\\d+\\.\\d+-\\d+\\.\\d+\\.\\d+(-SNAPSHOT)?\\.jar"
+  private val VERSIONS_REGEX = ".*hydra_(\\d+\\.\\d+\\.\\d+)-(\\d+\\.\\d+\\.\\d+(-SNAPSHOT)?)\\.jar".r
```

Comment:
Aren't VERSIONS_REGEX and HYDRA_REGEX going to match exactly the same text? Just wondering if both are really needed.

Reply:
Yes, you are right. Nice catch.